### PR TITLE
Clean up "supported environments" readme section

### LINKED ISSUE #52: Clean up "supported environments" readme section
Currently the "supported environments" section can be confusing as it states that we support UE 4.27 and 5.0.3, Windows 10 and macOS 12. It would be more applicable to mention that the supported operating systems for the plugin are:
- Windows 10+
- macOS:
  - UE 4.27: latest Mojave
  - UE 5: latest Monterey  
 
which are determined as being the highest common factor between UE and the C++ SDK (UE4 supports Windows 7, but SDK does not; SDK supports Mojave, but UE5 does not).

And that we support the following Unreal Engine versions: 4.27, 5.0, (soon) 5.1.

### PR DESCRIPTION
**Pull request by Rishav1707**
## This PR closes the following issue:
closes #52 

## Changes made:
- Supported operating systems for the plugin are:
```
- Windows 10+
- macOS:
       - UE 4.27: latest Mojave
       - UE 5: latest Monterey
```

### CODE REVIEW

**Comment by Rishav1707** [comment]
Unable to sign CLA @kubaau

**Comment by Rishav1707** [comment]
> Unable to sign CLA @kubaau

It gets stuck after clicking the sign-in button.

File: README.md [hunk 1]
```diff
@@ -2,8 +2,10 @@
 Plugin integrating [Dolby.io Communications](https://dolby.io) with the Unreal Engine.
 
 ## Supported environments
-- Unreal Engine 4.27.2 and 5.0.3
-- Windows 10 and macOS 12
+- Windows 10+
```
Comment (Adam1-3) [inline README.md:5]: I miss some info about UE versions for Windows e.g. `(all EU versions)`
```suggestion
- Windows 10+ (all UE versions)
```

**Review by Adam1-3 — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (kubaau) [reply README.md:5]: I think this should be worded differently. And it is certainly not "all UE versions" ;). Maybe something like in https://api-references.dolby.io/comms-sdk-cpp/other/supported_platforms.html

```
## Supported platforms
The plugin is compatible with the following Unreal Engine versions:
- 4.27.2
- 5.0.3

The plugin is compatible with the following operating systems:
- Windows 10, Windows 11
- macOS:
    - UE4: latest Mojave or newer
    - UE5: latest Monterey or newer
```

**Review by Adam1-3 — COMMENTED** [review]
We need to update link to SDK release in line 144 of README.md

**Review by kubaau — APPROVED** [review]
Approved except will slightly reword to remove mention of "all UE versions".

**Pull request merged**